tests/test_isgri_plugin.py

```diff
@@ -224,13 +254,88 @@ def test_isgri_lc_odaapi(dispatcher_live_fixture):
 
-    
+    #data = np.array(product_spiacs.spi_acs_lc_0_query.data_unit[1].data)
+    #assert len(data) > 100
 
 
+@pytest.mark.odaapi
+@pytest.mark.isgri_plugin
+def test_valid_token_oda_api(dispatcher_live_fixture):
+    import oda_api.api
 
-    #assert product_spiacs.spi_acs_lc_0_query.data_unit[1].header['INSTRUME'] == "SPI-ACS"
+    # let's generate a valid token
+    token_payload = {
+        **default_token_payload,
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
 
-    #data = np.array(product_spiacs.spi_acs_lc_0_query.data_unit[1].data)
-    #assert len(data) > 100
-    
\ No newline at end of file
+    disp = oda_api.api.DispatcherAPI(
+        url=dispatcher_live_fixture)
+    product = disp.get_product(
+        query_status="new",
+        product_type="Dummy",
+        instrument="empty",
+        product="dummy",
+        osa_version="OSA10.2",
+        E1_keV=40.0,
+        E2_keV=200.0,
+        scw_list=[f"0665{i:04d}0010.001" for i in range(200)],
+        token=encoded_token
+    )
+
+    logger.info("product: %s", product)
+    logger.info("product show %s", product.show())
+
+    session_id = disp.session_id
+    job_id = disp.job_id
+
+    # check query output are generated
+    query_output_json_fn = f'scratch_sid_{session_id}_jid_{job_id}/query_output.json'
+    # the aliased version might have been created
+    query_output_json_fn_aliased = f'scratch_sid_{session_id}_jid_{job_id}_aliased/query_output.json'
+    assert os.path.exists(query_output_json_fn) or os.path.exists(query_output_json_fn_aliased)
+    # get the query output
+    if os.path.exists(query_output_json_fn):
+        f = open(query_output_json_fn)
+    else:
+        f = open(query_output_json_fn_aliased)
+
+    jdata = json.load(f)
+
+    assert jdata["status_dictionary"]["debug_message"] == ""
+    assert jdata["status_dictionary"]["error_message"] == ""
+    assert jdata["status_dictionary"]["message"] == ""
+
+
+@pytest.mark.odaapi
+@pytest.mark.isgri_plugin
+@pytest.mark.xfail
+def test_invalid_token_oda_api(dispatcher_live_fixture):
+    import oda_api.api
+
+    # let's generate an expired token
+    exp_time = int(time.time()) - 500
+    # expired token
+    token_payload = {
+        **default_token_payload,
+        "exp": exp_time
+    }
+    encoded_token = jwt.encode(token_payload, secret_key, algorithm='HS256')
+
+    disp = oda_api.api.DispatcherAPI(
+        url=dispatcher_live_fixture)
+    # with raises_if_failing("failing", Exception):
+    with pytest.raises(oda_api.api.RemoteException):
+        product = disp.get_product(
+            query_status="new",
+            product_type="Dummy",
+            instrument="empty",
+            product="dummy",
+            osa_version="OSA10.2",
+            E1_keV=40.0,
+            E2_keV=200.0,
+            scw_list=[f"0665{i:04d}0010.001" for i in range(5)],
+            token=encoded_token
+        )
```

Comment:
these two tests are of generic and could be don

I rather meant tests which sets some acutal integral-specific roles and check that role validation is done accordingly. We will also then complete role application with these tests.

Maybe please make for me a placeholder where specific domain implementation is not clear?

we need to check-complete how two first roles from here are applied:

https://github.com/oda-hub/doc-multi-user/blob/main/plan-roles-users.md

Reply:
I had indeed in mind to have those two generic tests in the dispatcher-app.

Whereas here, having role specific tests

Reply:
So, I would move those two somewhere else, and go ahead with the role-tests implementation.

What do you think?

Reply:
exactly!